core/lib/zksync_core/src/eth_watch/mod.rs

```diff
@@ -131,13 +157,15 @@ impl EthWatch {
         }
     }
 
-    pub async fn run(
+    async fn run(
         &mut self,
         pool: ConnectionPool,
         stop_receiver: watch::Receiver<bool>,
+        limit: Option<usize>,
```

Comment:
Bikeshedding: I feel it would be better to stop the component via `stop_receiver` by either watching its side-effects (that is, the storage), or by having the component emit events that can be subscribed to. One way to achieve the latter is to add `watch::Sender<bool>` to `SetChainIDEventProcessor` (initially `false`; set to `true` once at least one event is received). This would mean you don't need to change `EventProcessor` signature etc.